pkg/runtime.v2/framework/plugins/plainml/plainml.go

```diff
@@ -40,16 +42,34 @@ func (p *PlainML) Name() string {
 	return Name
 }
 
-func (p *PlainML) EnforceMLPolicy(info *runtime.Info) error {
-	if info == nil || info.MLPolicy == nil || info.MLPolicy.Torch != nil || info.MLPolicy.MPI != nil {
+func (p *PlainML) EnforceMLPolicy(info *runtime.Info, trainJob *kubeflowv2.TrainJob, runtimeMLPolicy *kubeflowv2.MLPolicy) error {
+	if info == nil || runtimeMLPolicy == nil || runtimeMLPolicy.Torch != nil || runtimeMLPolicy.MPI != nil {
 		return nil
 	}
-	numNodes := ptr.Deref(info.MLPolicy.NumNodes, 1)
+
+	// TrainJob contains the actual information for the number of nodes.
+	numNodes := runtimeMLPolicy.NumNodes
+	if trainJob.Spec.Trainer != nil && trainJob.Spec.Trainer.NumNodes != nil {
+		numNodes = trainJob.Spec.Trainer.NumNodes
+	}
+	info.Trainer.NumNodes = numNodes
```

Comment:
@andreyvelich should we check if elastic policy is defined before setting numNodes?

Reply:
I think numNodes should default to 1 in the controller as we are checking on a condition with cel and can't be defaulted to 1 with kubebuilder markers - https://github.com/kubeflow/training-operator/pull/2313/files#diff-f18910f437a912a2683dddc8a4df118f70849e297f0e0f07961141967f482491R150

Reply:
Elastic Policy can be applied only in the Torch Plugin and I added TODO to implement it: https://github.com/kubeflow/training-operator/blob/85d0cf1fca0e56d206e579f77d68f33aca1a0f87/pkg/runtime.v2/framework/plugins/torch/torch.go#L51

Can we make default `numNodes=1` only if elastic policy is not set in `torch` API ?

Reply:
Can we make default `numNodes=1` only if elastic policy is not set in `torch` API ?

Reply:
Can we just open ElasticPolicy supporting issue, then support all Torch Elastic behaviors (defaulting and propagate the elastic parameters to JobSet) in a single issue?

Reply:
Yes, I will open separate issues for all TODOs after this PR.